### PR TITLE
Syntax highlighting fixes

### LINKED ISSUE #53: Syntax highlighting
For Scala and R support, we may want to look at http://pygments.org/languages/

For SQL support, look at @alope107's solution in https://github.com/alope107/spark-sql-magic/commit/34b4d864cdaaa0cc56481b3045a66bf679141f78

### LINKED ISSUE #187: Scala syntax highlighting?
Syntax highlighting in Scala is goofy.  It only highlights some cells -- looks like it highlights the first cell you execute but no other cell after that?  This is a regression in Jupyter's text editor or something.

image

### PR DESCRIPTION
**Pull request by msftristew**
Closes #187 

These changes seem to fix all the syntax highlighting issues we have observed recently.  On my Windows machine, syntax highlighting works fine across multiple cells and there are no additional Javascript errors in the console.

### CODE REVIEW

**Comment by MohamedElKamhawy** [comment]
Build fails?

File: remotespark/kernels/sparkkernel/sparkkernel.py [hunk 1]
```diff
@@ -13,6 +13,7 @@ def __init__(self, **kwargs):
         language_info = {
             'name': 'scala',
             'mimetype': 'text/x-scala',
+            'codemirror_mode': 'text/x-scala',
```
Comment (aggFTW) [inline remotespark/kernels/sparkkernel/sparkkernel.py]: Was this a regression? Why did it not fail before?

File: remotespark/kernels/sparkkernel/kernel.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+define(['base/js/namespace'], function(IPython){
+        var onload = function() {
+            IPython.CodeCell.config_defaults.highlight_modes['magic_text/x-sql'] = {'reg':[/^%%sql/]};
```
Comment (aggFTW) [inline remotespark/kernels/sparkkernel/kernel.js]: How did you test this works?

**Comment by msftristew** [comment]
Sorry build fails, I didn't know there was a unit test corresponding to this. Will fix in a moment

File: remotespark/kernels/sparkkernel/sparkkernel.py (hunk 1 above)
Comment (msftristew) [inline remotespark/kernels/sparkkernel/sparkkernel.py]: This is not a regression, this line was never present in the file.

I am unsure why it didn't fail before, though this seems to be the correct way to set the kernel's language to Scala.

File: remotespark/kernels/sparkkernel/kernel.js (hunk 2 above)
Comment (msftristew) [inline remotespark/kernels/sparkkernel/kernel.js]: * Reinstalled kernels, opened new notebooks in PySpark and Spark

* Created multiple cells starting with %%sql to verify that syntax highlighting worked in every cell starting with %%sql

* Checked Javascript console to ensure there were no unexpected errors

**Comment by msftristew** [comment]
NVM, according to further investigation, `%%sql` syntax highlighting still isn't 100% functional yet :(

**Comment by msftristew** [comment]
This is related to autoviz. When I turn autoviz off, SQL syntax highlighting works as expected in subsequent cells, but it fails when autoviz is turned back on.

**Comment by aggFTW** [comment]
Could it be related to https://github.com/plotly/plotly.js/issues/90?

**Comment by msftristew** [comment]
Unsure. Also wondering if this [this commit](https://github.com/plotly/plotly.py/commit/127ebc5f0be979c52847a6427a2ada3b11fc4352) is related at all.  Going to continue to investigate and try to get a minimal repro.

**Comment by msftristew** [comment]
Have a repro, going to file an issue against plotly to see what we can do about this

**Comment by msftristew** [comment]
What I'm probably going to do here is remove the SQL syntax highlighting code from this PR and then we can add it back in when the path forward on this is clear.

**Comment by aggFTW** [comment]
That sounds like a good compromise.

**Pull request merged**